frame/contracts/src/lib.rs

```diff
@@ -629,13 +634,15 @@ decl_module! {
 			let mut gas_meter = GasMeter::new(gas_limit);
 			let result = Self::execute_wasm(origin, &mut gas_meter, |ctx, gas_meter| {
 				let executable = PrefabWasmModule::from_storage(code_hash, &ctx.schedule)?;
+				let code_len = executable.pristine_size();
```

Comment:
why do we use pristine_size instead of instrumented code len ?
the value queried from storage is the instrumented one, wouldn't it make more sense to use its len instead ?

Reply:
This is because it is very difficult to create a benchmark that creates an instrumented code with a specific size. We assume a linear relationship between pristine and instrumented code size.  Now that I think about it I realized that my benchmarks are actually wrong because they do not instrument the code.

Reply:
Fixed it:
* Now using the instrumented size
* Fail code deployment if the instrumented code is above the max code size